core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -716,10 +721,10 @@ class KafkaController(val config: KafkaConfig,
     // check if they are already completed or topic was deleted
     val partitionsThatCompletedPreferredReplicaElection = partitionsUndergoingPreferredReplicaElection.filter { partition =>
       val replicas = controllerContext.partitionReplicaAssignment(partition)
-      val topicDeleted = replicas.isEmpty
-      val successful =
-        if (!topicDeleted) controllerContext.partitionLeadershipInfo(partition).leaderAndIsr.leader == replicas.head else false
-      successful || topicDeleted
+      if (replicas.isEmpty)
+        false // topic was deleted
```

Comment:
Shouldn't we return `true`? In the old logic, `replicas.isEmpty == true` => `topicDeleted == true` => `successful || topicDeleted == true`.